src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -575,6 +600,7 @@ public final SPEC metrics(boolean metricsEnabled) {
 				}
 			}
```

Comment:
This code is obsolete because stakeholders can use metering system different than Micrometer even if Micrometer is in the class path.

Reply:
This code is indeed obsolete if we accept that the `registrar` might be null in the `metrics(boolean metricsEnabled, Supplier<? extends ConnectionProvider.MeterRegistrar> registrar)` method, in which case it will end up with the default configuration being settled in the `acquire` method of the `ConnectionProvider`. 

I'm also not sure that the boolean `metricsEnabled` as a parameter is relevant: at the opposite of the `HttpClient.metrics` method, in case the boolean value is false, nothing has to be removed of the configuration. So we might simply give it `false` as a default value and pass it to true only when the method is called.

Reply:
> This code is indeed obsolete if we accept that the `registrar` might be null in the `metrics(boolean metricsEnabled, Supplier<? extends ConnectionProvider.MeterRegistrar> registrar)` method, in which case it will end up with the default configuration being settled in the `acquire` method of the `ConnectionProvider`.
> 

We can skip this check for `metrics(boolean metricsEnabled, Supplier<? extends ConnectionProvider.MeterRegistrar> registrar)`, while for `metrics(boolean metricsEnabled)` we definitely need it. This is similar to what we have for `HttpClient.metrics...`
https://github.com/reactor/reactor-netty/blob/d8f7acc1eb00775f7f2eb7681c8c8cb987296b62/src/main/java/reactor/netty/http/client/HttpClient.java#L1109-L1130
https://github.com/reactor/reactor-netty/blob/d8f7acc1eb00775f7f2eb7681c8c8cb987296b62/src/main/java/reactor/netty/http/client/HttpClient.java#L1159-L1163
I was thinking for something like this:

We keep the current `metrics(boolean metricsEnabled)` as it is at the moment, `registrar` is null and we will apply the default in `acquire`

```
public final SPEC metrics(boolean metricsEnabled) {
	if (metricsEnabled) {
		if (!Metrics.isInstrumentationAvailable()) {
			throw new UnsupportedOperationException(
					"To enable metrics, you must add the dependency `io.micrometer:micrometer-core`" +
							" to the class path first");
		}
	}
	this.metricsEnabled = metricsEnabled;
	return get();
}
```

And introduce a new method like this:

```
public final SPEC metrics(boolean metricsEnabled, Supplier<? extends ConnectionProvider.MeterRegistrar> registrar) {
	this.metricsEnabled = metricsEnabled;
	this.registrar = Objects.requireNonNull(registrar, "registrar");
	return get();
}
```

> I'm also not sure that the boolean `metricsEnabled` as a parameter is relevant: at the opposite of the `HttpClient.metrics` method, in case the boolean value is false, nothing has to be removed of the configuration. So we might simply give it `false` as a default value and pass it to true only when the method is called.

`metricsEnabled` is relevant and it is checked in `acquire` method.
`metricsEnabled ` is controlled by the user, see comment https://github.com/reactor/reactor-netty/pull/1218#discussion_r459555510